src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +173,241 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
+                # This will only happen if something goes wrong in our metadata
+                if hub_model_version > model_version:
+                    pass
+
+                # 3. Old model version exists in Hub, update
+                if hub_model_version < model_version:
+                    # Check minSDKVersion against current SDK version, emit log
+                    models_to_sync.append(model)
+
+        return models_to_sync
+
+    def sync(self, model_list: List[Dict[str, str]]):
+        """Syncs a list of JumpStart model ids and versions with a CuratedHub
+
+        Args:
+            model_list (List[Dict[str, str]]): List of `{ model_id: str, version: Optional[str] }`
+                objects that should be synced into the Hub.
+        """
+        if self._is_invalid_model_list_input(model_list):
+            raise ValueError(
+                "Model list should be a list of objects with values 'model_id',",
+                "and optional 'version'.",
+            )
+
+        # Retrieve latest version of unspecified JumpStart model versions
+        model_version_list = []
+        for model in model_list:
+            version = model.get("version", "*")
+            if version == "*":
+                model = self._populate_latest_model_version(model)
+                JUMPSTART_LOGGER.warning(
+                    "No version specified for model %s. Using version %s",
+                    model["model_id"],
+                    model["version"],
+                )
+            model_version_list.append(JumpStartModelInfo(model["model_id"], model["version"]))
+
+        js_models_in_hub = self._get_jumpstart_models_in_hub()
+
+        models_to_sync = self._determine_models_to_sync(model_version_list, js_models_in_hub)
+        JUMPSTART_LOGGER.warning(
+            "Syncing the following models into Hub %s: %s", self.hub_name, models_to_sync
+        )
+
+        # Delete old models?
+
+        # CopyContentWorkflow + `SageMaker:ImportHubContent` for each model-to-sync in parallel
+        tasks: List[futures.Future] = []
+        with futures.ThreadPoolExecutor(
+            max_workers=self._default_thread_pool_size,
+            thread_name_prefix="import-models-to-curated-hub",
+        ) as deploy_executor:
+            for thread_num, model in enumerate(models_to_sync):
+                task = deploy_executor.submit(self._sync_public_model_to_hub, model, thread_num)
+                tasks.append(task)
+
+        # Handle failed imports
+        results = futures.wait(tasks)
+        failed_imports: List[Dict[str, Any]] = []
+        for result in results.done:
+            exception = result.exception()
+            if exception:
+                failed_imports.append(
+                    {
+                        "Exception": exception,
+                        "Traceback": "".join(
+                            traceback.TracebackException.from_exception(exception).format()
```

Comment:
We will want to be descriptive about what model id and version each exception is related to. In MH, we see this issue where you get a dump of stacktraces without knowing which model each one relates to.